### PR TITLE
applications: serial_lte_modem: update sample.yaml

### PR DESCRIPTION
**Pull request by eivindj-nordic**
Update sample.yaml 
NCSDK-23316

### CODE REVIEW

**Review by tomi-font — APPROVED** [review]
Thanks!

File: applications/serial_lte_modem/sample.yaml [hunk 1]
```diff
@@ -1,15 +1,15 @@
 sample:
-  name: serial lte modem sample
+  name: Serial LTE Modem application
 tests:
-  applications.nrf9160.serial_lte_modem:
```
Comment (greg-fer) [inline applications/serial_lte_modem/sample.yaml:4]: Do you also want to remove nRF9160 from the name of the app in the docs?
I see 9161 is in platform_allow, so having nRF9160 in the app name might be a thing outdated.
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/applications/serial_lte_modem/README.html

Comment (eivindj-nordic) [reply applications/serial_lte_modem/sample.yaml:4]: I'll rename this to `nRF91 Series`.

Comment (greg-fer) [reply applications/serial_lte_modem/sample.yaml:4]: Ah, but #12120 uses just "Serial LTE modem", so I think @eivindj-nordic that "nRF91 Series" in the doc title is not needed.

File: applications/serial_lte_modem/sample.yaml [hunk 2]
```diff
@@ -1,15 +1,15 @@
 sample:
-  name: serial lte modem sample
+  name: Serial LTE Modem application
```
Comment (greg-fer) [inline applications/serial_lte_modem/sample.yaml:2]:
```suggestion
  name: Serial LTE Modem
```

File: applications/serial_lte_modem/README.rst [hunk 3]
```diff
@@ -1,7 +1,7 @@
 .. _serial_lte_modem:
 
-nRF9160: Serial LTE modem
-#########################
+nRF91 Series: Serial LTE modem
+##############################
```
Comment (greg-fer) [inline applications/serial_lte_modem/README.rst:4]:
```suggestion
Serial LTE modem
################
```
To match #12120 .

**Comment by greg-fer** [comment]
FYI, @wiba-nordic .

**Comment by wiba-nordic** [comment]
I think that since we already have a PR for changing the docs part, better to make all of those changes there and keep this one just for the yaml file.

**Comment by eivindj-nordic** [comment]
> I think that since we already have a PR for changing the docs part, better to make all of those changes there and keep this one just for the yaml file.

Agreed, I'll drop the changes to the docs for this one.

**Review by wiba-nordic — APPROVED** [review]

**Review by MarkusLassila — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Pull request merged**